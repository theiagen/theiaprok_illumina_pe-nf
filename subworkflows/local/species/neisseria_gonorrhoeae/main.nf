include { NGMASTER } from '../../../../modules/local/species_typing/neisseria/ngmaster/main'

workflow NEISSERIA_GONORRHOEAE_TYPING {

    take:
    ch_ng_assembly

    main:
    ch_ngmaster_results = Channel.empty()
    ch_versions = Channel.empty()

    NGMASTER (ch_ng_assembly)
    ch_ngmaster_results = NGMASTER.out.ngmast_report
    ch_versions = ch_versions.mix(NGMASTER.out.versions)

    emit:
    ngmast_report = ch_ngmaster_results
    versions = ch_versions
}