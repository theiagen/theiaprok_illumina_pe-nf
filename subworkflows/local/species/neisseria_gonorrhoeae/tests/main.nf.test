nextflow_workflow {

    name "Test Subworkflow NEISSERIA_GONORRHOEAE_TYPING"
    script "../main.nf"
    workflow "NEISSERIA_GONORRHOEAE_TYPING"

    test("neisseria_gonorrhoeae - test") {
        setup {
            run("GUNZIP") {
                script "../../../../../modules/nf-core/gunzip/main.nf"
                process {
                    """
                    input[0] = [
                        [id:'test_sample'],
                        file('https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/019/048/825/GCF_019048825.1_ASM1904882v1/GCF_019048825.1_ASM1904882v1_genomic.fna.gz', checkIfExists: true)
                    ]
                    """
                }
            }
        }
        when {
            workflow {
                """
                input[0] = GUNZIP.out.gunzip
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.ngmast_report,
                    workflow.out.versions
                ).match() }
            )
        }
    }
}