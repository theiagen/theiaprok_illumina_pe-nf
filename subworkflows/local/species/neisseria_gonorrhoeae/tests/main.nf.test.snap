{
    "neisseria_gonorrhoeae - test": {
        "content": [
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample.ngmaster.tsv:md5,4a9c6bfd0dbd47db52198c39bd18b428"
                ]
            ],
            [
                "versions.yml:md5,b5a0289d69f6d663696a023e5f10b89a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.8"
        },
        "timestamp": "2025-07-02T21:23:51.576298329"
    }
}